src/main/java/org/jenkinsci/plugins/docker/workflow/FromFingerprintStep.java

```diff
@@ -87,8 +89,9 @@ public String getToolName() {
             String fromImage = null;
             FilePath dockerfile = workspace.child(step.dockerfile);
             InputStream is = dockerfile.read();
+            BufferedReader r = null;
             try {
-                BufferedReader r = new BufferedReader(new InputStreamReader(is)); // encoding probably irrelevant since image/tag names must be ASCII
+                r = new BufferedReader(new InputStreamReader(is, "ASCII")); // encoding probably irrelevant since image/tag names must be ASCII
```

Comment:
whilst a the tag or name that is within this file **may** be limited to the characters that are available within US-ASCII unless I have missed something there is no restrictions on the file not being in whatever encoding is supported by your system, so the characters may require some mapping from one charset to another.

e.g. this file *could* be written using UTF-16 and as such the line we are looking for could be 
"F□R□O□M□ □M□y□T□a□g□" and as such we would not find this line.